### PR TITLE
Fix to check traceparent header in gRPC metadata for grpc-dotnet client

### LINKED ISSUE #1824: Add traceparent header in gRPC metadata for grpc-dotnet client
# Description

grpc-dotnet client adheres to OpenTelemetry Spec which only supports http based traceparent header in gRPC calls.

Dapr dotnet-SDK to use W3C Trace Context feature built in ASP.NET Core and .NET Core 3.0 + , need to add workaround fix to have `traceparent` header in gRPC response as grpc-trace-bin is not yet there in OpenTelemetry unlike OpenCensus .  

## Issue reference

https://github.com/dapr/dapr/issues/1711
https://github.com/dapr/dotnet-sdk/issues/300

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Adding workaround fix to have `traceparent` header in gRPC response as grpc-trace-bin is not yet there in OpenTelemetry unlike OpenCensus .  

grpc-dotnet client adheres to OpenTelemetry Spec which only supports http based traceparent header in gRPC calls.

This will enable Dapr dotnet-SDK to use W3C Trace Context feature built in ASP.NET Core and .NET Core 3.0 + .

## Issue reference
Closes #1824 
https://github.com/dapr/dapr/issues/1711
https://github.com/dapr/dotnet-sdk/issues/300

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by shalabhms** [comment]
Tested E2E with Dapr dotnet SDK.

image

**Comment by yaron2** [comment]
So how short is the term, really?

File: pkg/diagnostics/tracing.go [hunk 1]
```diff
@@ -70,6 +70,19 @@ func SpanContextToW3CString(sc trace.SpanContext) string {
 		[]byte{byte(sc.TraceOptions)})
 }
 
+// TraceStateToW3CString extracts the TraceState from given SpanContext and returns its string representation
+func TraceStateToW3CString(sc trace.SpanContext) string {
```
Comment (youngbupark) [inline pkg/diagnostics/tracing.go:74]: please add this unit-test.

File: pkg/diagnostics/grpc_tracing.go [hunk 2]
```diff
@@ -122,6 +122,17 @@ func SpanContextFromIncomingGRPCMetadata(ctx context.Context) (trace.SpanContext
 	if len(traceContext) > 0 {
 		traceContextBinary := []byte(traceContext[0])
 		sc, ok = propagation.FromBinary(traceContextBinary)
+	} else {
+		// add workaround to fallback on checking traceparent header
```
Comment (youngbupark) [inline pkg/diagnostics/grpc_tracing.go:126]: I would add TODO: comment to recognize the code later.

File: pkg/diagnostics/tracing.go (hunk 1 above)
Comment (shalabhms) [reply pkg/diagnostics/tracing.go:74]: done

File: pkg/diagnostics/grpc_tracing.go (hunk 2 above)
Comment (shalabhms) [reply pkg/diagnostics/grpc_tracing.go:126]: done.

**Pull request merged**